dask/dataframe/core.py

```diff
@@ -645,6 +645,14 @@ def assign(self, **kwargs):
 
         return elemwise(_assign, self, *pairs, columns=list(df2.columns))
 
+    @wraps(pd.DataFrame.rename)
+    def rename(self, **kwargs):
+        index = kwargs.get('index')
+        columns = kwargs.get('columns')
+        func = getattr(self._partition_type, 'rename')
+        return map_partitions(func, self.column_info, self, index, columns)
```

Comment:
Should `column_info` be changed here?

I usually construct a dummy `pd.DataFrame` and then run that dataframe through the same function with the same args.

In [1]: import dask.dataframe as dd
dd
In [2]: import pandas as pd

In [3]: df = dd.from_pandas(pd.DataFrame({'x': [1, 2, 3, 4]}), npartitions=2)

In [4]: df
Out[4]: dd.DataFrame<from_pandas-1, divisions=(0, 2, 3)>

In [5]: df.columns
Out[5]: ('x',)

In [7]: df.rename(columns={'x': 'X'}).columns
Out[7]: ('x',)

In [8]: df.rename(columns={'x': 'X'}).compute()
Out[8]: 
   X
0  1
1  2
2  3
3  4
```

This is also a sign that perhaps our `eq` function for testing needs to be changed.